### PR TITLE
Chore: Restore automated checks after codebase grafting

### PR DESCRIPTION
**Pull request by kjohnson**
## Description

This PR makes a number of updates/changes to resolve various discrepancies in the automation, including tests and static analysis.

## Affects

- Now required PHP (7.2)
- Now requires WordPress (6.0).
- PHP Money does not support all currencies.
- Changed instances of `new static` to prevent unsafe usage.
- Added stubs for static analysis of WP CLI functions.

### CODE REVIEW

**Comment by kjohnson** [comment]
I've got PHP Compatibility and Static Analysis resolved, but now there are some failing tests related to currency.

File: .github/workflows/php-compatibility.yml [hunk 1]
```diff
@@ -9,3 +9,5 @@ on:
 jobs:
     build:
         uses: impress-org/givewp-github-actions/.github/workflows/php-compatibility.yml@master
+        with:
+            test_versions: "[ '7.2', '8.0' ]"
```
Comment (JasonTheAdams) [inline .github/workflows/php-compatibility.yml:13]: Let's remember to bump the shared workflow to 7.2 after 3.0 is released.

File: src/DonationForms/DataTransferObjects/AuthenticationData.php [hunk 2]
```diff
@@ -23,7 +23,7 @@ class AuthenticationData
      */
     public static function fromRequest(array $request): AuthenticationData
     {
-        $self = new static();
+        $self = new AuthenticationData();
```
Comment (JasonTheAdams) [inline src/DonationForms/DataTransferObjects/AuthenticationData.php:26]:
```suggestion
        $self = new self();
```

File: src/DonationForms/DataTransferObjects/DonateFormRouteData.php [hunk 3]
```diff
@@ -45,7 +45,7 @@ class DonateFormRouteData implements Arrayable
      */
     public static function fromRequest(array $requestData): DonateFormRouteData
     {
-        $self = new static();
+        $self = new DonateFormRouteData();
```
Comment (JasonTheAdams) [inline src/DonationForms/DataTransferObjects/DonateFormRouteData.php:48]:
```suggestion
        $self = new self();
```

File: src/DonationForms/DataTransferObjects/DonateRouteData.php [hunk 4]
```diff
@@ -30,7 +30,7 @@ class DonateRouteData
      */
     public static function fromRequest(array $request): DonateRouteData
     {
-        $self = new static();
+        $self = new DonateRouteData();
```
Comment (JasonTheAdams) [inline src/DonationForms/DataTransferObjects/DonateRouteData.php:33]:
```suggestion
        $self = new self();
```

File: src/DonationForms/DataTransferObjects/DonationConfirmationReceiptViewRouteData.php [hunk 5]
```diff
@@ -20,7 +20,7 @@ class DonationConfirmationReceiptViewRouteData
      */
     public static function fromRequest(array $request): DonationConfirmationReceiptViewRouteData
     {
-        $self = new self();
+        $self = new DonationConfirmationReceiptViewRouteData();
```
Comment (JasonTheAdams) [inline src/DonationForms/DataTransferObjects/DonationConfirmationReceiptViewRouteData.php:23]: ...What's the difference?

**Review by JasonTheAdams — COMMENTED** [review]
I noticed we're changing `return static` to explicit classes? Why not use `return self`? We also have a return type changing from `self` to the class name. I don't think that's the issue, it's just the use of `static`.

File: src/DonationForms/DataTransferObjects/DonationConfirmationReceiptViewRouteData.php (hunk 5 above)
Comment (kjohnson) [reply src/DonationForms/DataTransferObjects/DonationConfirmationReceiptViewRouteData.php]: There was mixed usage between `self` and the explicit class name, so I just picked a direction. I don't think that there is a difference, in practice.

File: .github/workflows/php-compatibility.yml (hunk 1 above)
Comment (kjohnson) [reply .github/workflows/php-compatibility.yml:13]: How should we track that todo?

File: src/FormMigration/Actions/TransferDonations.php [hunk 6]
```diff
@@ -2,12 +2,27 @@
...
+class TransferDonations
 {
+    protected $sourceId;
+
+    public function __construct($sourceId)
+    {
+        $this->sourceId = $sourceId;
+    }
+
+    public static function from($sourceId): TransferDonations
```
Comment (JasonTheAdams) [inline src/FormMigration/Actions/TransferDonations.php:16]:
```suggestion
    public static function from($sourceId): self
```

File: src/FormMigration/Actions/TransferDonations.php [hunk 7]
```diff
@@ -2,12 +2,27 @@
...
+    {
+        return new TransferDonations($sourceId);
+    }
+
+    public function to($destinationId)
```
Comment (JasonTheAdams) [inline src/FormMigration/Actions/TransferDonations.php:21]:
```suggestion
    public function to($destinationId): void
```

File: src/FormMigration/Actions/TransferFormUrl.php [hunk 8]
```diff
@@ -2,10 +2,25 @@
...
+class TransferFormUrl
 {
+    protected $sourceId;
+
+    public function __construct($sourceId)
+    {
+        $this->sourceId = $sourceId;
+    }
+
+    public static function from($sourceId): TransferFormUrl
```
Comment (JasonTheAdams) [inline src/FormMigration/Actions/TransferFormUrl.php:14]:
```suggestion
    public static function from($sourceId): self
```

File: src/FormMigration/Actions/TransferFormUrl.php [hunk 9]
```diff
@@ -2,10 +2,25 @@
...
+    {
+        return new TransferFormUrl($sourceId);
+    }
+
+    public function to($destinationId)
```
Comment (JasonTheAdams) [inline src/FormMigration/Actions/TransferFormUrl.php:19]:
```suggestion
    public function to($destinationId): void
```

File: src/FormMigration/DataTransferObjects/FormMigrationPayload.php [hunk 10]
```diff
@@ -19,8 +19,8 @@ public function __construct(DonationFormV2 $formV2, DonationFormV3 $formV3)
         $this->formV3 = $formV3;
     }
 
-    public static function fromFormV2(DonationFormV2 $formV2)
+    public static function fromFormV2(DonationFormV2 $formV2): FormMigrationPayload
```
Comment (JasonTheAdams) [inline src/FormMigration/DataTransferObjects/FormMigrationPayload.php:22]:
```suggestion
    public static function fromFormV2(DonationFormV2 $formV2): self
```

File: src/FormMigration/DataTransferObjects/FormMigrationPayload.php [hunk 11]
```diff
@@ -19,8 +19,8 @@ public function __construct(DonationFormV2 $formV2, DonationFormV3 $formV3)
...
     {
-        return new static($formV2, DonationFormV3::factory()->create());
+        return new FormMigrationPayload($formV2, DonationFormV3::factory()->create());
```
Comment (JasonTheAdams) [inline src/FormMigration/DataTransferObjects/FormMigrationPayload.php:24]:
```suggestion
        return new self($formV2, DonationFormV3::factory()->create());
```

File: src/FormMigration/DataTransferObjects/TransferOptions.php [hunk 12]
```diff
@@ -22,18 +22,18 @@ public function __construct(bool $changeUrl, bool $delete, bool $redirect)
         $this->redirect = $redirect;
     }
 
-    public static function fromRequest(WP_REST_Request $request)
+    public static function fromRequest(WP_REST_Request $request): TransferOptions
```
Comment (JasonTheAdams) [inline src/FormMigration/DataTransferObjects/TransferOptions.php:25]:
```suggestion
    public static function fromRequest(WP_REST_Request $request): self
```

File: src/FormMigration/DataTransferObjects/TransferOptions.php [hunk 13]
```diff
@@ -22,18 +22,18 @@ public function __construct(bool $changeUrl, bool $delete, bool $redirect)
...
     {
-        return new static(
+        return new TransferOptions(
```
Comment (JasonTheAdams) [inline src/FormMigration/DataTransferObjects/TransferOptions.php:27]:
```suggestion
        return new self(
```

File: src/FormMigration/DataTransferObjects/TransferOptions.php [hunk 14]
```diff
@@ -22,18 +22,18 @@ public function __construct(bool $changeUrl, bool $delete, bool $redirect)
...
             $request->get_param('changeUrl'),
             $request->get_param('delete'),
             $request->get_param('redirect')
         );
     }
 
-    public static function fromArray($options)
+    public static function fromArray($options): TransferOptions
```
Comment (JasonTheAdams) [inline src/FormMigration/DataTransferObjects/TransferOptions.php:34]:
```suggestion
    public static function fromArray($options): self
```

File: src/FormMigration/DataTransferObjects/TransferOptions.php [hunk 15]
```diff
@@ -22,18 +22,18 @@ public function __construct(bool $changeUrl, bool $delete, bool $redirect)
...
     {
-        return new static(
+        return new TransferOptions(
```
Comment (JasonTheAdams) [inline src/FormMigration/DataTransferObjects/TransferOptions.php:36]:
```suggestion
        return new self(
```

**Review by JasonTheAdams — CHANGES_REQUESTED** [review]
Some more `self` suggestions. I think you can bulk accept.

**Review by JasonTheAdams — APPROVED** [review]
Looks good to me, @kjohnson! If everything passes let's merge it! Thanks for owning this and taking care of it!

**Pull request merged**